execution-engine/contracts/hdac-system/pop/src/pop_contract.rs

```diff
@@ -281,6 +283,12 @@ impl ProofOfProfessionContract {
     }
 
     pub fn write_genesis_total_supply(&self, genesis_total_supply: &U512) -> Result<()> {
+        let caller = runtime::get_caller();
+
+        if caller.value() != consts::SYSTEM_ACCOUNT) {
```

Comment:
typo.

Everything looks good except for this. 🙂

Reply:
:sweat_smile: :+1: